### PR TITLE
Site: add `Env` page

### PR DESCRIPTION
**Pull request by iRevive**

### CODE REVIEW

**Review by djspiewak — DISMISSED** [review]

File: docs/std/env.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  } yield ()  
+```
+
+### Why `def get(name): F[Option[String]]` rather than `def get(name): Option[String]`?
+
+Technically, environment variables can be set within the **current** process on a POSIX system.  
+Java doesn't provide that functionality in a public API, but it's possible via JNA API or in Scala Native world.
+
+Since the library supports multiple platforms (JVM, Scala Native, ScalaJS), the implementation should take
+platform-specific peculiarities into the account.
```
Comment (armanbilge) [inline docs/std/env.md]: Also, I think we can get a security error on the JVM, and we would need an effect to capture that.

Comment (iRevive) [reply docs/std/env.md]: Indeed. I will add this detail tomorrow.

Comment (armanbilge) [reply docs/std/env.md]: Also, I just remembered: the AWS Lambda runtime actually _does_ change env variables while the process is running.

https://github.com/aws/aws-lambda-java-libs/blob/04ef611147fba47202c376da56741bf857b1edba/aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/AWSLambda.java#L267-L271

Comment (iRevive) [reply docs/std/env.md]: "Wake up, samurai. We have _env variables_ to change."

Now I'm 100% convinced it should be effectful.

Comment (armanbilge) [reply docs/std/env.md]: We even rely on this in natchez :)
https://github.com/tpolecat/natchez/blob/ffa7ef08d40cb1dcf4a27e4e28ff2d66c45b3aa1/modules/xray/src/main/scala/natchez/xray/XRayEnvironment.scala#L33-L34

Comment (iRevive) [reply docs/std/env.md]: I updated the paragraph.

**Review by armanbilge — APPROVED** [review]
Thanks for all these docs PRs! :)

**Pull request merged**